app/Console/Commands/VerifyInfinitePayPayments.php

```diff
@@ -0,0 +1,36 @@
+namespace App\Console\Commands;
+
+use Illuminate\Console\Command;
+use Illuminate\Support\Facades\Http;
+use App\Models\Invoice;
+
+class VerifyInfinitePayPayments extends Command
+{
+    protected $signature = 'infinitepay:verify-payments';
+    protected $description = 'Verifica pagamentos confirmados na InfinitePay e baixa faturas automaticamente';
+
+    public function handle()
+    {
+        $invoices = Invoice::where('status_id', Invoice::STATUS_SENT)
+            ->whereNotNull('custom_value1') // transaction_id
+            ->whereNotNull('custom_value2') // slug
+            ->get();
+
+        foreach ($invoices as $invoice) {
+            $gateway = $invoice->company->company_gateways()->first();

```

Comment:
You'll want to filter by gateways with this gateway_key, this may or may not hit the required gateway.